sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ModelsRepositoryClient.cs

```diff
@@ -90,27 +92,29 @@ public ModelsRepositoryClient(Uri repositoryUri, ModelsRepositoryClientOptions o
         }
 
         /// <summary>
-        /// Resolves a model definition identified by <paramref name="dtmi"/> and optionally its dependencies.
+        /// Gets a model definition identified by <paramref name="dtmi"/> and optionally its dependencies.
         /// </summary>
         /// <returns>
         /// An IDictionary containing the model definition(s) where the key is the dtmi
         /// and the value is the raw model definition string.
         /// </returns>
         /// <exception cref="RequestFailedException">Thrown when a resolution failure occurs.</exception>
         /// <param name="dtmi">A well-formed DTDL model Id. For example 'dtmi:com:example:Thermostat;1'.</param>
+        /// <param name="resolutionOption">A DependencyResolutionOption value to force model resolution behavior.</param>
         /// <param name="cancellationToken">The cancellationToken.</param>
         [SuppressMessage(
             "Usage",
             "AZC0015:Unexpected client method return type.",
             Justification = "Item lookup is optimized with a dictionary type, we do not expect any more than ~20 items to be returned. TODO: azabbasi: discuss this issue with the review board.")]
-        public virtual IDictionary<string, string> Resolve(string dtmi, CancellationToken cancellationToken = default)
+        public virtual IDictionary<string, string> GetModels(
+            string dtmi, DependencyResolutionOption? resolutionOption = null, CancellationToken cancellationToken = default)
         {
-            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ModelsRepositoryClient)}.{nameof(Resolve)}");
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ModelsRepositoryClient)}.{nameof(GetModels)}");
             scope.Start();
 
             try
             {
-                return _repositoryHandler.Process(dtmi, cancellationToken);
+                return _repositoryHandler.Process(dtmi, resolutionOption, cancellationToken);
```

Comment:
We determine which resolution option should be passed down here, and make it required (validation failure) down the stream.

Reply:
Done.